src/lsp_data.rs

```diff
@@ -291,3 +291,34 @@ where
         }
     }
 }
+
+/* ----------  Temporary LSP type until window/progress proposal is done --------- */
+
+/**
+ * The progress notification is sent from the server to the client to ask the client
+ * to indicate progress.
+ */
+#[allow(non_upper_case_globals)]
+pub const NOTIFICATION__Progress: &'static str = "window/progress";
+
+#[derive(Debug, PartialEq, Deserialize, Serialize, Clone)]
+pub struct ProgressParams {
+    // A unique identifier to associate multiple progress notifications with the same progress.
+    pub id: String,
+
+    // The title of the progress.
+    // This should be the same for all ProgressParams with the same id.
+    pub title: Option<String>,
+
+    // Optional progress message to display.
+    // If unset, the previous progress message (if any) is still valid.
+    pub message: Option<String>,
```

Comment:
It'd be good to add `#[serde(skip_serializing_if = "Option::is_none")]` here not to send unset/irrelevant values to client

Reply:
Fair enough